src/ws/service/helpers.rs

```diff
@@ -0,0 +1,456 @@
+use std::sync::Arc;
```

Comment:
Can we use a more meaningful name for this file? And if it's necessary to create one file per scope that'd be great.

Reply:
Sure! I thought of this: one file contains functions for working with the database, another file contains functions for working with Synapse, and the last file contains utility functions for mapping and extracting data.

Reply:
If you have any suggestions or concerns, please let me know by commenting! :)